test/spec/controllers/tasksCtrlSpec.js

```diff
@@ -30,11 +30,89 @@ describe('Tasks Controller', function() {
   });
 
   describe('editTask', function() {
-    it('is Tasks.editTask', function() {
-      inject(function(Tasks) {
-        expect(scope.editTask).to.eql(Tasks.editTask);
-      });
+
+    var task;
+
+    beforeEach(function(){
+      task = specHelper.newTask();
+    });
+
+    it('toggles the _editing property', function() {
+      scope.editTask(task, user);
+      expect(task._editing).to.eql(true);
+    });
+
+    it('sets _tags to true by default', function() {
+      scope.editTask(task, user);
+
+      expect(task._tags).to.eql(true);
+    });
+
+    it('sets _tags to false if preference for collapsed tags is turned on', function() {
+      user.preferences.tagsCollapsed = true;
+      scope.editTask(task, user);
+
+      expect(task._tags).to.eql(false);
+    });
+
+    it('sets _advanced to true by default', function(){
+      user.preferences.advancedCollapsed = true;
+      scope.editTask(task, user);
+
+      expect(task._advanced).to.eql(false);
+    });
+
+    it('sets _advanced to false if preference for collapsed advance menu is turned on', function() {
+      user.preferences.advancedCollapsed = false;
+      scope.editTask(task, user);
+
+      expect(task._advanced).to.eql(true);
     });
+
+
+  });
+
+  describe('cancelEdit', function() {
+
+    var task;
+
+    beforeEach(function(){
+      task = specHelper.newTask();
+    });
+
+    it('resets the task text', function() {
+      let originalText = task.text;
+      scope.editTask(task, user);
+      task.text = 'test';
```

Comment:
Exactly!  The sanity check is to make sure you aren't "changing" a string to the exact same string.